crates/router/src/core/pm_auth.rs

```diff
@@ -66,6 +67,19 @@ pub async fn create_link_token(
 
     let pm_auth_key = format!("pm_auth_{}", payload.payment_id);
 
+    redis_conn
+        .exists::<Vec<u8>>(&pm_auth_key)
+        .await
+        .change_context(ApiErrorResponse::InvalidRequestData {
+            message: "Incorrect payment_id provided in request".to_string(),
+        })
+        .attach_printable("Corresponding pm_auth_key does not exist in redis")?
```

Comment:
This should be `InternalServerError` as the first error propagation is due to redis error. If the bool is `false` then it should be 4xx.

Can be taken in the next PR.